subiquity/client/controllers/mirror.py

```diff
@@ -27,13 +29,29 @@ class MirrorController(SubiquityTuiController):
 
     async def make_ui(self):
         mirror = await self.endpoint.GET()
-        return MirrorView(self, mirror)
+        has_network = await self.app.client.network.has_network.GET()
+        if has_network:
+            check = await self.endpoint.check_mirror.progress.GET()
+        else:
+            check = None
```

Comment:
Feels like the check for network could be done server side inside progress.GET() but OK.

Reply:
I have the impression that we have more and more screens that behave differently based on the check for network and so far the implementation has been pretty inconsistent:

* GET /drivers returns a "local_only" boolean that tells if we should look online or only in the media
* GET /ubuntu_pro returns a "has_network" boolean
* GET /snaplist returns SnapCheckState.FAILED if no network is available

so I tried to define a consistent way for the client to get that information.